src/garage/torch/policies/gaussian_mlp_policy.py

```diff
@@ -0,0 +1,139 @@
+            > std_parameterization
+                'exp' : plain value (not log or exponentiated)
+                'softplus' : softplus value
+        std_hidden_nonlinearity: Nonlinearity for each hidden layer in
+            the std network.
+        std_output_nonlinearity: Nonlinearity for output layer in
+            the std network.
+        std_parametrization (str): How the std should be parametrized. There
+            are a few options:
+        - exp: the logarithm of the std will be stored, and applied a
+            exponential transformation
+        - softplus: the std will be computed as log(1+exp(x))
+        layer_normalization (bool): Bool for using layer normalization or not.
+    :return:
+
+    """
+    def __init__(self,
+                 env_spec,
+                 hidden_sizes=(32, 32),
+                 hidden_nonlinearity=torch.tanh,
+                 hidden_w_init=nn.init.xavier_uniform_,
+                 hidden_b_init=nn.init.zeros_,
+                 output_nonlinearity=None,
+                 output_w_init=nn.init.xavier_uniform_,
+                 output_b_init=nn.init.zeros_,
+                 learn_std=True,
+                 adaptive_std=False,
+                 std_share_network=False,
+                 init_std=1.0,
+                 min_std=1e-6,
+                 max_std=None,
+                 std_hidden_sizes=(32, 32),
+                 std_hidden_nonlinearity=torch.tanh,
+                 std_output_nonlinearity=None,
+                 std_parameterization='exp',
+                 layer_normalization=False):
+        nn.Module.__init__(self)
+        Policy.__init__(self, env_spec)
+
+        obs_dim = env_spec.observation_space.flat_dim
+        action_dim = env_spec.action_space.flat_dim
+
+        self._model = GaussianMLPModule(
+            input_dim=obs_dim,
+            output_dim=action_dim,
+            hidden_sizes=hidden_sizes,
+            hidden_nonlinearity=hidden_nonlinearity,
+            hidden_w_init=hidden_w_init,
+            hidden_b_init=hidden_b_init,
+            output_nonlinearity=output_nonlinearity,
+            output_w_init=output_w_init,
+            output_b_init=output_b_init,
+            learn_std=learn_std,
+            adaptive_std=adaptive_std,
+            std_share_network=std_share_network,
+            init_std=init_std,
+            min_std=min_std,
+            max_std=max_std,
+            std_hidden_sizes=std_hidden_sizes,
+            std_hidden_nonlinearity=std_hidden_nonlinearity,
+            std_output_nonlinearity=std_output_nonlinearity,
+            std_parameterization=std_parameterization,
+            layer_normalization=layer_normalization
+        )
+
+    def forward(self, inputs):
+        return self._model(inputs)
+
+    def get_action(self, observation):
+        dist = self.forward(observation)
+
+        return (dist.rsample()[0].detach().numpy(),
+                dict(
+                    mean=dist.mean[0].detach().numpy(),
```

Comment:
this style of returning the mean and log_std in get_action is an outdated part of TF which we are removing. please return only the action.

Reply:
removed